jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt

```diff
@@ -0,0 +1,426 @@
+        val newSelectedEndpointID = message.selectedEndpoint
+        val newSelectedIDs: List<String> =
+            if (newSelectedEndpointID == null || StringUtils.isBlank(newSelectedEndpointID)) emptyList()
+            else listOf(newSelectedEndpointID)
+        selectedEndpoints(SelectedEndpointsMessage(newSelectedIDs))
+        return null
+    }
+
+    /**
+     * Handles an opaque message received on the Relay channel. The message originates from an endpoint with an ID of
+     * `message.getFrom`, as verified by the remote bridge sending the message.
+     *
+     * @param message the message that was received from the endpoint.
+     */
+    override fun endpointMessage(message: EndpointMessage): BridgeChannelMessage? {
+        // We trust the "from" field, because it comes from another bridge, not an endpoint.
+        val conference = relay.conference
+        if (conference.isExpired) {
+            logger.warn("Unable to send EndpointMessage, conference is expired")
+            return null
+        }
+        val targets = if (message.isBroadcast()) {
+            // Broadcast message
+            conference.localEndpoints
+        } else {
+            // 1:1 message
+            val to = message.to
+            val targetEndpoint = conference.getLocalEndpoint(to)
+            if (targetEndpoint != null) {
+                listOf(targetEndpoint)
+            } else {
+                logger.warn("Unable to find endpoint to send EndpointMessage to: $to")
+                return null
+            }
+        }
+        conference.sendMessage(message, targets, false /* sendToOcto */)
+        return null
+    }
+
+    /**
+     * Handles an endpoint statistics message on the Octo channel that should be forwarded to
+     * local endpoints as appropriate.
+     *
+     * @param message the message that was received from the endpoint.
+     */
+    override fun endpointStats(message: EndpointStats): BridgeChannelMessage? {
+        // We trust the "from" field, because it comes from another bridge, not an endpoint.
+        val conference = relay.conference
+        if (conference.isExpired) {
+            logger.warn("Unable to send EndpointStats, conference is null or expired")
+            return null
+        }
+        if (message.from == null) {
+            logger.warn("Unable to send EndpointStats, missing from")
+            return null
+        }
+        val from = conference.getEndpoint(message.from!!)
+        if (from == null) {
+            logger.warn("Unable to send EndpointStats, unknown endpoint " + message.from)
+            return null
+        }
+        val targets = conference.localEndpoints.stream()
+            .filter { ep: Endpoint ->
+                ep.wantsStatsFrom(
+                    from
+                )
+            }
+            .collect(Collectors.toList())
+        conference.sendMessage(message, targets, false)
```

Comment:
Do you think it's necessary to send the message via `Conference`? I think I did that at some point, but I don't remember if there was a good reason anymore. It would be simpler to just do `localEndpoints.filter { ...}.forEach { it.sendMessage(message) }`